core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -1038,6 +1093,9 @@ private[kafka] class Processor(val id: Int,
    * Close the selector and all open connections
    */
   private def closeAll(): Unit = {
+    // Clear to unblock blocked acceptors
+    newConnections.asScala.foreach(_.close())
```

Comment:
The blocked acceptor would then add another connection to this list right? Do we close that one?

Reply:
No, we don't close that one. Let me rework this.